### PR TITLE
feature/string char literals

### PR DESCRIPTION
**Pull request by ozansz**
### String Literal Grammar
**string_literal**: `'"' (single_nonescape_char | escape_sequence)* '"'`
**single_nonescape_char**: Single character from all printable characters except '"' (double quote), '\n' (newline) and '\' (backslash)
**escape_sequence**: `'\' (common_string_escape | ascii_string_escape | unicode_string_escape)`
**common_string_escape**: `(n | r | t | 0)`
**ascii_string_escape**: `x[0-9A-Fa-f]{2}`
**unicode_string_escape**: `u[0-9A-Fa-f]{4}`

### Char Literal Grammar
**string_literal**: `''' (single_nonescape_char | escape_sequence) '''`

### CODE REVIEW

**Comment by oznakn** [comment]
Looks nice!

**Comment by umut-sahin** [comment]
**unicode_string_escape** should be able to take up to 6 hex digits.

File: external/tree-sitter-pvl/grammar.js [hunk 1]
```diff
@@ -69,8 +69,31 @@ module.exports = grammar({
             field("value", alias($.decimal_number, $.floating_point_number)),
             field("type_hint", $.floating_point_type_hint),
           ),
+
+          field("value", $.character_literal),
+          field("value", $.string_literal),
```
Comment (umut-sahin) [inline external/tree-sitter-pvl/grammar.js]: These rules should not have **_literal** suffixes for consistency with the rest of the grammar.

File: external/tree-sitter-pvl/grammar.js [hunk 2]
```diff
@@ -69,8 +69,31 @@ module.exports = grammar({
...
         ),
 
+				character_literal: $ =>
+          seq(
+            '\'',
```
Comment (umut-sahin) [inline external/tree-sitter-pvl/grammar.js]: All the strings in the grammar should be wrapped between double quotes.
I know it's silly but hey, CONSISTENCY! :smile:

File: external/tree-sitter-pvl/grammar.js [hunk 3]
```diff
@@ -69,8 +69,31 @@ module.exports = grammar({
...
+            field("value", alias(choice(
+              $.single_nonescape_char,
+              $.escape_sequence
```
Comment (umut-sahin) [inline external/tree-sitter-pvl/grammar.js]: Also it should be impossible to have unicode escape sequence in a char literal.

**Pull request closed without merging**